src/target/source/codegen_opencl.cc

```diff
@@ -472,20 +445,15 @@ void CodeGenOpenCL::VisitExpr_(const CallNode* op, std::ostream& os) {
     this->PrintExpr(op->args[2], ss);
     ss << ")))";
 
-    // Only use local SSA if texture is not already being stored
-    if (need_texture_ssa_) {
-      std::string rhs = SSAGetID(ss.str(), op->dtype.with_lanes(4));
-      if (op->args.back().as<RampNode>()) {
-        os << rhs;
-      } else {
-        os << "((";
-        this->PrintType(op->dtype.with_lanes(1), os);
-        os << "*)&" << rhs << ")[";
-        this->PrintExpr(op->args.back(), os);
-        os << "]";
-      }
```

Comment:
IIRC, without checking if the data is already stored, this ended up repeated stores, resulting in less optimal opencl code. It might be optimized by the runtime compiler but I recall a non-trivial perf difference.

Reply:
Understood, if you find a case with this in the future please consider adding it as a follow up test. Otherwise all LGTM, thanks @echuraev!